xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -123,14 +128,44 @@ bool CVAAPIContext::EnsureContext(CVAAPIContext **ctx, CDecoder *decoder)
   return true;
 }
 
+namespace
+{
+int find_open_render_node_fd()
```

Comment:
> but reading about render nodes they are independent of cards ie. any should work but I cant test that

Assume you have a system with multiple cards. You would have a render node for every device. Does not matter as long you do not intend to bring the vaSurface to display. But in case you do want to display it, you need to find the render node for the device the app has its display on.